### DISCUSSION TITLE
Appliance connection logging ?

### DISCUSSION THREAD

**Discussion by YannickG1** [camueller/SmartApplianceEnabler#475] [Installation und Betrieb]
Hello,

I just started trying out SAE because my home manager 2.0 got installed yesterday. First of all, thank you already for your hard work! I was wondering if I can activate any logging about the meter/control HTTP connection ? It is not working (not doing anything) and I don't know why because the log is empty.. I'm running latest docker image (so 2.1.0) and configured the http connection for both meter and counter. It is a local API with easy json format that shows kWh used. Example:

GET http://192.168.1.234/api/v1/data returns this:
`{"wifi_ssid":"Moms-Spaghetti","wifi_strength":34,"total_power_import_t1_kwh":407.551,"total_power_export_t1_kwh":0,"active_power_w":59.013,"active_power_l1_w":59.013}`

Configured like this:
<img width="992" alt="meter-setting" src="https://user-images.githubusercontent.com/3669955/219790394-35d099df-1dcc-4870-8913-f0806cbaf460.png">

Then it needs to control and recognise when my dishwasher is starting so that it can turn it off and wait for better moment (triggered by home manager). I configured that like this:

<img width="1103" alt="control-setting" src="https://user-images.githubusercontent.com/3669955/219790792-9d1587c8-40a6-46f5-81e4-fab93530dd07.png">

I have read all the documentation and it seems to be okay.. but my dishwasher is consuming 60W at the moment and the Status traffic light just remains black the whole time, while it should have recognised that it is running.

That's why I want to ask:
1) is there a way to enable more verbose logging to see like the http requests SAE is making?
2) If there is a communication error, should I normally see it in the logfile / console output ? Because there  are no errors printed there..

Thanks!

Edit: At the same time I’m trying to figure out why the api of my smart socket becomes unresponsive when SAE has been running for about 5-10 minutes. It almost looks like SAE is ddos’ing it or something because immediately after I stop the docker, the socket api becomes available again.

**Comment by camueller** [comment]
> is there a way to enable more verbose logging to see like the http requests SAE is making?

https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Logging_EN.md#log-level

> If there is a communication error, should I normally see it in the logfile / console output ? Because there are no errors printed there..

Communication errors should always show up in the log file regardless of log level.

>  It almost looks like SAE is ddos’ing it or something because immediately after I stop the docker, the socket api becomes available again.

You didn't mention that you are running the Docker container. In my experience most problems stem from the fact that people don't understand containerization in general and/or the containerization tool they are using. I would recommend to get SAE running without Docker first and the move the configuration to a Docker container.

**Comment by YannickG1** [comment]
Hello @camueller ,

Ok my bad, I was expecting SAE to log to stdout in the docker version so that the logs are visible in the docker console log.. I logged into the container and indeed found the rolling file in the tmp folder. After looking into the subject again today, I think I have found more or less the issue. The processing power of the smart socket seems to be very low and it does not like sending a lot of headers in the request. I know from the past that java adds some headers by default etc so what I did is put a Nginx proxy in front of the api of the socket and force nginx to remove all headers in the request + cache the response for 5 seconds so that the api cannot get overloaded.

Calling the nginx proxy to get the data instead of directly to the local API of the socket completely solved the issue. The local api of the socket no longer becomes unavailable after a few minutes of running SAE, and SAE correctly identifies a requested load from my dishwasher after about 10 seconds of turning it on (yellow traffic light visible and planning request visible in the semp endpoint).

One last question then but it is more about home manager.. How long until a load planning should be visible for my device . I see in the sunnyportal logs that it connected to the SAE  "semp gateway" and identified the  dishwasher  device but nothing is showing up yet on the load planning page when it would activate the dishwasher.

Thanks!

**Comment by camueller** [comment]
> How long until a load planning should be visible for my device . I see in the sunnyportal logs that it connected to the SAE "semp gateway" and identified the dishwasher device but nothing is showing up yet on the load planning page when it would activate the dishwasher.

It may take up to 15 minutes.

**Comment by camueller** [comment]
Out of curiosity: In which country do you live?

Good to see some english speaking users :-)
Only 2 moths ago I added an English translation of the documentation and fixed some issues with English in the web user interface as well.

**Comment by YannickG1** [comment]
I'm from Belgium :) I believe SMA is mostly popular in Western Europe. I used Chrome translate function in the beginning before I saw the english versions of the doc. Did the same to get EVCC running, works good enough :D 

Still cannot get forecast planning to work unfortunately :( It just shows a thin line (like there is no load planning request I assume).

SEMP output:
`This XML file does not appear to have any style information associated with it. The document tree is shown below.
<Device2EM xmlns="http://www.sma.de/communication/schema/SEMP/v1">
<DeviceInfo>
<Identification>
<DeviceId>F-19901204-120419900001-00</DeviceId>
<DeviceName>FSK63607P</DeviceName>
<DeviceType>DishWasher</DeviceType>
<DeviceSerial>10510010</DeviceSerial>
<DeviceVendor>AEG</DeviceVendor>
</Identification>
<Characteristics>
<MaxPowerConsumption>2200</MaxPowerConsumption>
</Characteristics>
<Capabilities>
<CurrentPower>
<Method>Measurement</Method>
</CurrentPower>
<Timestamps>
<AbsoluteTimestamps>false</AbsoluteTimestamps>
</Timestamps>
<Interruptions>
<InterruptionsAllowed>false</InterruptionsAllowed>
</Interruptions>
<Requests>
<OptionalEnergy>true</OptionalEnergy>
</Requests>
</Capabilities>
</DeviceInfo>
<DeviceStatus>
<DeviceId>F-19901204-120419900001-00</DeviceId>
<EMSignalsAccepted>true</EMSignalsAccepted>
<Status>Off</Status>
<PowerConsumption>
<PowerInfo>
<AveragePower>0</AveragePower>
<Timestamp>0</Timestamp>
<AveragingInterval>60</AveragingInterval>
</PowerInfo>
</PowerConsumption>
</DeviceStatus>
<PlanningRequest>
<Timeframe>
<DeviceId>F-19901204-120419900001-00</DeviceId>
<EarliestStart>0</EarliestStart>
<LatestEnd>95652</LatestEnd>
<MinRunningTime>8999</MinRunningTime>
<MaxRunningTime>9000</MaxRunningTime>
</Timeframe>
</PlanningRequest>
</Device2EM>`

<img width="1103" alt="loadplanning" src="https://user-images.githubusercontent.com/3669955/219894897-574a9574-8f27-4ea2-aa84-8bac1818f83c.png">

Not sure it  is SAE fault to be honest since I just started using home manager. Might be a configuration issue in home manager  as well of course  :) 

If I force the start of the dishwasher via SAE, I do see wattage consumption numbers change every 60 seconds in sunny portal monitoring graph so the connection between home manager and SAE seems to be ok, otherwise it would not recognise the wattage changes in sunny portal when it is actually running

**Comment by camueller** [comment]
The SEMP-XML looks good and should cause the Sunny Home Manager to plan a runtime for the dish washer.

Since you are using Docker my first guess would be that the Sunny Home Manager is not really communicating the SAE:
https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/QA_EN.md#semp1

If communication is working you might reset the Sunny Home Manager in order to clean its caches:
- SHM 2.0: Press reset button longer than 6 seconds
- SHM 1.0. Disconect power, wait some seconds, reconnect power
